go/vt/vtgate/evalengine/eval_tuple.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package evalengine
+
+import (
+	"vitess.io/vitess/go/sqltypes"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+type evalTuple struct {
+	t []eval
+}
+
+func (e *evalTuple) hash() (HashCode, error) {
+	return 0, vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "cannot hash TUPLE")
```

Comment:
why? can't we just merge the individual hashes using some clever algo with prime numbers?

Reply:
This is a wrong error message; the hash method for tuple should never be called because tuple-equality comparison is always done sequentially in the evalengine (to follow MySQL's approach), so the hashes of two tuples should never be compared. `Code_UNIMPLEMENTED` is definitely wrong here. I'll update this to make it more explicit.